paasta_tools/kubernetes/application/controller_wrappers.py

```diff
@@ -216,37 +217,31 @@ def create(self, kube_client: KubeClient) -> None:
         self.sync_horizontal_pod_autoscaler(kube_client)
 
     def deep_delete_and_create(self, kube_client: KubeClient) -> None:
```

Comment:
Are there any tests that need updating?

Reply:
Existing tests were unaffected, but I've added another to make sure we check this new logic